File: clojure/rna-transcription/src/rna_transcription.clj
(ns rna-transcription)

(def nucleotide->rna {\G \C, \C \G, \T \A, \A \U})

(defn to-rna
  "Returns the RNA complement of the given DNA string sequence."
  [dna]
  (apply str (map nucleotide->rna dna)))

File: python/rna-transcription/rna_transcription.py
"""Determine the RNA complement of a given DNA sequence"""

TRANSLATION = {'G': 'C', 'C': 'G', 'T': 'A', 'A': 'U'}

def to_rna(dna_strand):
    return ''.join(TRANSLATION[char] for char in dna_strand)
